nmrfx-processor/src/main/java/org/nmrfx/processor/processing/Processor.java

```diff
@@ -1435,9 +1435,14 @@ public void runProcesses() throws IncompleteProcessException {
             dataset.writeHeader(false);
         }
         if (!keepDatasetOpen) {
-            dataset.setNFreqDims(nDimsProcessed);
+            int freqDimsProcessed = 0;
+            for (int i = 0; i< dataset.getNDim(); i++) {
+                if (dataset.getFreqDomain(i)) {
+                    freqDimsProcessed++;
+                }
+            }
+            dataset.setNFreqDims(freqDimsProcessed);
             for (int i = nDimsProcessed; i < dataset.getNDim(); i++) {
-                dataset.setFreqDomain(i, false);
```

Comment:
Not a hundred percent sure on this change or how to test, but it seems like the freqDomain is being set correctly by the FT/IFT operations, so not sure why we are setting it here, it would be good to have an example of when nDimsProcessed would be less than the dataset dimensions

Reply:
nDimsProcessed would be less than dataset dimensions for experiments like t1/t2 pseudo-3D, but I think its correct that that line is not necessary